### PR TITLE
add check for win32 or msys before running rustc-with-gold fixes #9499

### LINKED ISSUE #9499: The gold linker should not be used on Windows
After https://github.com/servo/servo/pull/9449 building on Windows failed:
```
$ ./mach build -r
Could not execute process `C:/msys64/home/Urban/servo/etc/rustc-with-gold -vV` (never executed)

Caused by:
  %1 is not a valid Win32 application. (os error 193)
Build completed in 0.14s
```

### PR DESCRIPTION
**Pull request by jasonwilliams**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9588)

### CODE REVIEW

**Comment by Jayflux** [comment]
for https://github.com/servo/servo/issues/9499

File: python/servo/command_base.py [hunk 1]
```diff
@@ -379,7 +379,8 @@ def build_env(self, gonk=False, hosts_file_path=None):
 
         env['RUSTDOC'] = path.join(self.context.topdir, 'etc', 'rustdoc-with-private')
 
-        if self.config["tools"]["rustc-with-gold"]:
+        # Don't run the gold linker if on Windows https://github.com/servo/servo/issues/9499
+        if self.config["tools"]["rustc-with-gold"] and (sys.platform != "win32" and sys.platform != "msys"):
```
Comment (frewsxcv) [inline python/servo/command_base.py]: Nit: parenthesis are superfluous

FYI, you can write this like:

```python
if self.config["tools"]["rustc-with-gold"] and sys.platform not in ("win32" , "msys"):
```

Comment (Jayflux) [inline python/servo/command_base.py]: thanks @frewsxcv will update later

File: python/servo/command_base.py [hunk 2]
```diff
@@ -379,7 +379,8 @@ def build_env(self, gonk=False, hosts_file_path=None):
 
         env['RUSTDOC'] = path.join(self.context.topdir, 'etc', 'rustdoc-with-private')
 
-        if self.config["tools"]["rustc-with-gold"]:
+        # Don't run the gold linker if on Windows https://github.com/servo/servo/issues/9499
+        if self.config["tools"]["rustc-with-gold"] and sys.platform not in ("win32" , "msys"):
```
Comment (Wafflespeanut) [inline python/servo/command_base.py]: Travis caught you! "whitespace" before comma :)

**Comment by larsbergstrom** [comment]
Hrm, a new intermittent?
```
Tests with unexpected results:
  ▶ TIMEOUT [expected PASS] /css-backgrounds-3_dev/html4/border-bottom-left-radius-009.htm

  ▶ TIMEOUT [expected PASS] /css-backgrounds-3_dev/html4/border-radius-005.htm
```

cc @mbrubeck

**Comment by Wafflespeanut** [comment]
Heyyyyy... that's two new intermittents! O_o

**Comment by Jayflux** [comment]
@larsbergstrom  Whats that about?

**Comment by larsbergstrom** [comment]
@bors-servo retry
- #9626

**Pull request merged**